src/Advection/weno_fifth_order_stretched.jl

```diff
@@ -0,0 +1,339 @@
+#####
+##### Weighted Essentially Non-Oscillatory (WENO) scheme of order 5
+#####
+
+using OffsetArrays
+using Oceananigans.Grids: with_halo
+using Oceananigans.Architectures: arch_array
+using Adapt
+import Base: show
+
+const AG = AbstractGrid
+const two_32 = Int32(2)
+
+const C3₀ = 3/10
+const C3₁ = 3/5
+const C3₂ = 1/10
+
+const ƞ = Int32(2) # WENO exponent
+const ε = 1e-6
+
+struct WENO5{FT, XT, YT, ZT, Buffer} <: AbstractUpwindBiasedAdvectionScheme{2} 
+    
+    coeff_xᶠᵃᵃ::XT
+    coeff_xᶜᵃᵃ::XT
+    
+    coeff_yᵃᶠᵃ::YT
+    coeff_yᵃᶜᵃ::YT
+    
+    coeff_zᵃᵃᶠ::ZT
+    coeff_zᵃᵃᶜ::ZT
+ 
+end
+
+function WENO5(FT = Float64; grid = nothing) 
+    
+    if grid isa Nothing
+        coeff_xᶠᵃᵃ = nothing 
+        coeff_xᶜᵃᵃ = nothing  
+        coeff_yᵃᶠᵃ = nothing   
+        coeff_yᵃᶜᵃ = nothing
+        coeff_zᵃᵃᶠ = nothing
+        coeff_zᵃᵃᶜ = nothing
+    elseif !(typeof(grid) <: RectilinearGrid)
+        @warn "Stretched WENO is not supported with grids other than Rectilinear, defaulting to Uniform WENO"
```

Comment:
Is there a technical reason why this won't work on curvilinear grids, or is it merely something that needs to be implemented still?

Reply:
I still have to research how to change coefficients with a curvilinearly stretched grid (I guess the vertical direction would be fine)

Reply:
I missed that you did not implement the algorithm in [A simple algorithm to improve the performance of the WENO scheme on non-uniform grids](https://link.springer.com/article/10.1007/s10409-017-0715-2) (which reports results for curvilinear grids). Perhaps we will have to find another way then.